Raise handled exceptions

aljazkosir opened the pull request:
Addresses issue https://github.com/xlab-si/redfish-client-python/issues/8

\cc @miha-plesko @tadeboro

tadeboro commented on redfish_client/connector.py:
```diff
@@ -51,7 +51,11 @@ def _url(self, path):
+            resp = self._client.request(method, self._url(path), **args)
+        except requests.exceptions.ConnectionError:
+            raise InaccessibleException(
+                'Endpoint at {} is not accessible'.format(self._base_url))
```
Nitpick: We should use `"` in place of `'` for consistency with the rest of the code base.

aljazkosir commented:
@tadeboro I added unit tests for newly implemented exceptions and I also added unit tests for other exceptions that were missing them (I included them in this PR since we are refactoring exceptions, but I can also create new PR if wish)

tadeboro commented:
Sorry for the noise that I caused by closing and reopening this pull request. I misconfigured the CircleCI when I set up the integration. Closing and reopening triggered the build.

tadeboro commented:
Thank you for your contribution @aljazkosir.
